DQM/SiTrackerPhase2/plugins/Phase2ITMonitorRecHit.cc

```diff
@@ -0,0 +1,247 @@
+        layerMEs_[key].clusterSizeX->Fill(rechitIt->cluster()->sizeX());
+      if (layerMEs_[key].clusterSizeY)
+        layerMEs_[key].clusterSizeY->Fill(rechitIt->cluster()->sizeY());
+      if (layerMEs_[key].globalPosXY)
+        layerMEs_[key].globalPosXY->Fill(gx, gy);
+      if (layerMEs_[key].globalPosRZ)
+        layerMEs_[key].globalPosRZ->Fill(gz, gr);
+      if (layerMEs_[key].localPosXY)
+        layerMEs_[key].localPosXY->Fill(lp.x(), lp.y());
+      if (layerMEs_[key].posX)
+        layerMEs_[key].posX->Fill(lp.x());
+      if (layerMEs_[key].posY)
+        layerMEs_[key].posX->Fill(lp.y());
+      if (layerMEs_[key].posX)
+        layerMEs_[key].poserrX->Fill(rechitIt->localPositionError().xx());
+      if (layerMEs_[key].posY)
+        layerMEs_[key].poserrY->Fill(rechitIt->localPositionError().yy());
+    }  //end loop over rechits of a detId
+  }    //End loop over DetSetVector
+
+  //fill nRecHits per event
+  numberRecHits_->Fill(nTotrechitsinevt);
+  //fill nRecHit counter per layer
+  for (auto& lme : nrechitLayerMap)
+    if (layerMEs_[lme.first].numberRecHits)
+      layerMEs_[lme.first].numberRecHits->Fill(lme.second);
+}
+void Phase2ITMonitorRecHit::bookHistograms(DQMStore::IBooker& ibooker,
+                                           edm::Run const& iRun,
+                                           edm::EventSetup const& iSetup) {
+  std::string top_folder = config_.getParameter<std::string>("TopFolderName");
+  ibooker.cd();
+  edm::LogInfo("Phase2ITMonitorRecHit") << " Booking Histograms in : " << top_folder;
+  std::string dir = top_folder;
+  ibooker.setCurrentFolder(dir);
+  std::stringstream HistoName;
+  //Global histos for IT
+  HistoName.str("");
+  HistoName << "NumberRecHits";
+  numberRecHits_ = Phase2TkUtil::book1DFromPSet(
+      config_.getParameter<edm::ParameterSet>("GlobalNumberRecHits"), HistoName.str(), ibooker);
+  HistoName.str("");
+  HistoName << "Global_Position_XY_IT_barrel";
+  globalXY_barrel_ = Phase2TkUtil::book2DFromPSet(
+      config_.getParameter<edm::ParameterSet>("GlobalPositionXY_PXB"), HistoName.str(), ibooker);
+  HistoName.str("");
+  HistoName << "Global_Position_RZ_IT_barrel";
+  globalRZ_barrel_ = Phase2TkUtil::book2DFromPSet(
+      config_.getParameter<edm::ParameterSet>("GlobalPositionRZ_PXB"), HistoName.str(), ibooker);
+  HistoName.str("");
+  HistoName << "Global_Position_XY_IT_endcap";
+  globalXY_endcap_ = Phase2TkUtil::book2DFromPSet(
+      config_.getParameter<edm::ParameterSet>("GlobalPositionXY_PXEC"), HistoName.str(), ibooker);
+  HistoName.str("");
+  HistoName << "Global_Position_RZ_IT_endcap";
+  globalRZ_endcap_ = Phase2TkUtil::book2DFromPSet(
+      config_.getParameter<edm::ParameterSet>("GlobalPositionRZ_PXEC"), HistoName.str(), ibooker);
+
+  //Now book layer wise histos
+  edm::ESWatcher<TrackerDigiGeometryRecord> theTkDigiGeomWatcher;
+  edm::ESHandle<TrackerTopology> tTopoHandle;
+  iSetup.get<TrackerTopologyRcd>().get(tTopoHandle);
```

Comment:
this could ported to use esconsumes (also all other conditions handles everywhere in the PR).

Reply:
All of these now moved to esconsumes.